anchors/anchor_repository_integration_test.go

```diff
@@ -31,13 +34,68 @@ func TestMain(m *testing.M) {
 	os.Exit(result)
 }
 
-func TestCommitAnchor_Integration(t *testing.T) {
+func TestPreCommitAnchor_Integration(t *testing.T) {
 	anchorID := utils.RandomSlice(32)
+	signingRoot := utils.RandomSlice(32)
+
+	anchorIDTyped, _ := anchors.ToAnchorID(anchorID)
+	preCommitAnchor(t, anchorID, signingRoot)
+	valid := anchorRepo.HasValidPreCommit(anchorIDTyped)
+	assert.True(t, valid)
+}
+
+func TestPreCommit_CommitAnchor_Integration(t *testing.T) {
```

Comment:
this verifies that pre-commit -> commit flow works with contract, its just a matter of verifying coredoc processing flow. 
So I think smart contracts are ready for pre-audit now. @lucasvo @pstehlik @rdinicut